modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/Tabs.js

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayTabs from '@clayui/tabs';
+import React, {useEffect, useRef} from 'react';
+
+const Panel = ({children}) => {
+	const ref = useRef();
+
+	useEffect(() => {
+		ref.current.appendChild(children);
+	}, [children]);
+
+	return <div ref={ref}></div>;
```

Comment:
do we need a `ref` here? I started to doubt when I took a look to these examples: https://beta.reactjs.org/reference/react/Children#accepting-an-array-of-objects-as-a-prop (there is an example for tabs, very similar to this one...)

Reply:
You can't render a DOM node in a JSX, it's not an accepted type, that only works with React Components. If so, you get this error

```
Invariant Violation
Objects are not valid as a React child (found: [object HTMLDivElement]).
```

So we need to inject it directly in the DOM using appendChild.

[I already asked Matu](https://github.com/liferay/clay/issues/5368) to add a `forwardRef` in ClayTabs.Panel component so we don't need to add an extra container and use the component ref directly.